bgzf/writer.go

```diff
@@ -0,0 +1,116 @@
+package bgzf
+
+import (
+	"compress/gzip"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"log"
+	"math"
+)
+
+// magicHexEOF is a particular empty bgzf block that marks the true EOF.
+const magicHexEOF string = "1f8b08040000000000ff0600424302001b0003000000000000000000"
+
+// Writer moves data -> compressor -> zipBlock -> file writer
+type Writer struct {
+	w          io.Writer
+	compressor *gzip.Writer
+	zipBlock   *Block
+}
+
+// NewWriter creates a bgzf writer from any input writer.
+func NewWriter(w io.Writer) Writer {
+	var err error
+	var zw Writer
+	zw.w = w
+	zw.zipBlock = NewBlock()
+	zw.compressor = gzip.NewWriter(zw.zipBlock)
+	_, err = zw.compressor.Write([]byte{}) // write header
+	if err != nil {
+		log.Panic(err)
+	}
+	zw.zipBlock.Reset()
```

Comment:
I see the header write and the reset both here when initializing the writer and then again on Write commands.  Do we need them here?

Reply:
I think I added this here before I trashed the header on every write. This is no longer needed now that it is in the Write function and has been removed.